tests/test_terminal.py

```diff
@@ -362,34 +350,50 @@ def _(lhs=left, rhs=right, writer=writer):
         )
 
 
-pretty_registered_comparisons = (Comparison.In, Comparison.NotIn, Comparison.Equals)
-non_pretty_registered_comparisons = (
-    c for c in Comparison if c not in pretty_registered_comparisons
-)
+for comparison, description in [
+    (Comparison.Equals, "not equal to"),
+    (Comparison.NotEquals, "equal to"),
+    (Comparison.LessThan, "less than"),
+    (Comparison.LessThanEqualTo, "less than or equal to"),
+    (Comparison.GreaterThan, "greater than"),
+    (Comparison.GreaterThanEqualTo, "greater than or equal to"),
+]:
 
-for comparison in non_pretty_registered_comparisons:
+    @test(
+        "TestResultWriter.get_operands has a specialized description for `{comparison.value}`"
+    )
+    def _(writer=writer, comparison=comparison, description=description):
+        failure = TestFailure("fail", "a", "b", 1, comparison, "test")
+        lhs_render, rhs_render = writer.get_operands(failure).renderables
+
+        assert description in rhs_render.title.plain
+
+
+for comparison in Comparison:
 
     @test(
-        "TestResultWriter.get_pretty_comparison_failure handles unregistered comparison failures"
+        "TestResultWriter.get_pretty_comparison_failure can handle `{comparison.value}` failures"
     )
     def _(comparison=comparison, writer=writer):
-        failure = TestFailure("fail", "a", "b", 1, comparison, "test")
-        renderable: ConsoleRenderable = writer.get_pretty_comparison_failure(failure)
+        failure = TestFailure("fail", "a", "b", 1, comparison, "")
+        renderable = writer.get_pretty_comparison_failure(failure)
 
-        assert renderable == Text("", end="")
+        assert renderable is not None
 
 
 @test("TestResultWriter.output_all_test_results returns empty list if suite is empty")
 def _(console=mock_rich_console):
     suite = Suite([])
-    result_writer = TestResultWriter(
-        console=console,
-        suite=suite,
-        test_output_style=TestOutputStyle.TEST_PER_LINE,
-        progress_styles=[TestProgressStyle.INLINE],
-        config_path=None,
-    )
 
-    result = result_writer.output_all_test_results(_ for _ in ())
-    assert result == []
-    assert not console.print.called
+    with patch.object(console, "print") as mock:
+        result_writer = TestResultWriter(
+            console=console,
+            suite=suite,
+            test_output_style=TestOutputStyle.TEST_PER_LINE,
+            progress_styles=[TestProgressStyle.INLINE],
+            config_path=None,
+        )
+
+        result = result_writer.output_all_test_results(_ for _ in ())
+        assert result == []
+        assert not mock.called
```

Comment:
The old version of this test  looks correct and cleaner to me, unless I'm missing something here

Reply:
Woops - I had changed the `mock_rich_console` at one point, which necessitated this, but ended up reverting it, and didn't remember to revert this.